src/Serializer/src/Serializer/JsonSerializer.php

```diff
@@ -21,6 +22,12 @@ public function serialize(mixed $payload): string|\Stringable
      */
     public function unserialize(\Stringable|string $payload, object|string|null $type = null): mixed
     {
+        if ($type !== null) {
+            throw new UnserializeException(
```

Comment:
What about InvalidArgumentException that extends UnserializeException

Reply:
Done.